app/code/Magento/Catalog/Model/Indexer/Category/Product/AbstractAction.php

```diff
@@ -171,7 +172,7 @@ protected function reindex()
     {
         foreach ($this->storeManager->getStores() as $store) {
             if ($this->getPathFromCategoryId($store->getRootCategoryId())) {
-                $this->currentStoreId = $store->getId();
+                $this->currentStore = $store;
```

Comment:
better to move this to a method that ensures a strict type return
that way when getCurrentStore is used it can throw an error is currentStore is empty